pygmt/datasets/earth_mean_dynamic_topography.py

```diff
@@ -0,0 +1,104 @@
+"""
+Function to download the CNES Earth mean dynamic topography dataset from the GMT data
+server, and load as :class:`xarray.DataArray`.
+
+The grids are available in various resolutions.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_mean_dynamic_topography"]
+
+
+def load_earth_mean_dynamic_topography(
+    resolution: Literal["01d", "30m", "20m", "15m", "10m", "07m"] = "01d",
+    region: Sequence[float] | str | None = None,
+    registration: Literal["gridline", "pixel"] = "gridline",
+) -> xr.DataArray:
+    r"""
+    Load the CNES Earth mean dynamic topography dataset in various resolutions.
+
+    .. figure:: https://www.generic-mapping-tools.org/remote-datasets/_images/GMT_earth_mdt.jpg
+       :width: 80 %
+       :align: center
+
+       CNES Earth mean dynamic topography dataset.
+
+    The grids are downloaded to a user data directory (usually
+    ``~/.gmt/server/earth/earth_mdt/``) the first time you invoke this function.
+    Afterwards, it will load the grid from the data directory. So you'll need an
+    internet connection the first time around.
+
+    These grids can also be accessed by passing in the file name
+    **@earth_mdt**\_\ *res*\[_\ *reg*] to any grid processing function or plotting
+    method. *res* is the grid resolution (see below), and *reg* is the grid registration
+    type (**p** for pixel registration or **g** for gridline registration).
+
+    The default color palette table (CPT) for this dataset is *@earth_mdt.cpt*. It's
+    implicitly used when passing in the file name of the dataset to any grid plotting
+    method if no CPT is explicitly specified. When the dataset is loaded and plotted
+    as an :class:`xarray.DataArray` object, the default CPT is ignored, and GMT's
+    default CPT (*turbo*) is used. To use the dataset-specific CPT, you need to
+    explicitly set ``cmap="@earth_mdt.cpt"``.
+
+    Refer to :gmt-datasets:`earth-mdt.html` for more details about available datasets,
+    including version information and references.
+
+    Parameters
+    ----------
+    resolution
+        The grid resolution. The suffix ``d`` and ``m`` stand for arc-degrees and
+        arc-minutes. Note that ``"07m"`` refers to a resolution of 7.5 arc-minutes.
+    region
+        The subregion of the grid to load, in the form of a sequence [*xmin*, *xmax*,
+        *ymin*, *ymax*] or an ISO country code. Required for grids with resolutions
+        higher than 5 arc-minutes (i.e., ``"05m"``).
```

Comment:
No `05m` for this dataset:
```suggestion
        The subregion of the grid to load, in the form of a sequence [*xmin*, *xmax*,
        *ymin*, *ymax*] or an ISO country code.
```